packages/@aws-cdk/aws-codepipeline/lib/pipeline.ts

```diff
@@ -445,6 +415,93 @@ export class Pipeline extends PipelineBase {
     return actionRole;
   }
 
+  private getRoleFromActionPropsOrGenerateIfCrossAccount(stage: Stage, action: IAction): iam.IRole | undefined {
+    if (action.actionProperties.role) {
```

Comment:
Please add a comment before each "if" statement to explain the case and the behavior you are implementing.

Reply:
Done.